### ISSUE TITLE
The problem of remote working and Junior Developers

### ISSUE THREAD

**Issue by SegersIan** [BarcelonaJS/jobs#1]
# Preface #
This is a summary of a topic that was discussed in the BarcelonaJS Slack channel `#jobs`. I tried to summarize the things discussed in the channel. Feel free to comment, elaborate or correct my summary. I quoted some people If you would like your quote to be removed, feel free to ask me.

This issue was created after the initial suggestion of @PatrickHeneise .

# The Discussion #
The discussion was triggered by a request for a list of remote-first and/or remote-friendly companies based in Barcelona. This discussion topic converged into a topic on something along the lines of **Why it is important to have frequently Senior Developers in the office so Junior developers can benefit from this through proper training and facilities to learn**.

> remote-first should not ever be a priority but people first

Companies can tap into the massive pool of talent that is available, companies should, therefore, invest in their juniors to train them accordingly, having senior developers around is a must to achieve this.

An alternative for juniors or young talent to get proper training would be boot camps. The issue with boot camps is that they are not affordable for everyone. They can be very expensive (thousands of Euros). The majority of the Juniors cannot afford these prices, some code camps partner up for 'student loans' but they mostly only cover the tuition fee. This means that the student still needs extra money to cover their own living expenses. This seems to be rather an option for people coming from a specific 'income-class'.

The high tuition fee for code camps is also not improving inclusivity in the Tech sector. Any person from any financial, religious or ethnic background should have the possibility to grow into Tech.

**So what are the alternatives for facilitating the young developers or people who want to get into Tech out there?**

In Germany and Japan, it's common to do an apprenticeship, a 3-year training in a company on a low salary (600-900/month) and gets trained up in your field (i.e. Software Programmer). The only prerequisite is to have a middle school degree.

If anyone is interested in facilitating such apprenticeships, feel free to contact @PatrickHeneise 

>  I’m in contact with the German IHK (Chamber of Commerce and Trade) to make this happen in Spain

In addition to the apprenticeship concept, there are also 'after-office' hour meetups like

* JavaScript 101 Workshops
* NodeSchool
* FreeCodeCamp

** What to learn first **
After discussing the need for more developers and appropriate facilities to train them, we discussed also the ideal 'start' for new people starting in Tech. Should they start immediately with the full web stack (JS/HTML/CSS) or rather keep a narrow focus on just one (like only JS).

It was clear that there is a broad amount of opinions out there, but let me list a few that were pointed out:

> we need developers, not markup people. So JS first and only, there’s plenty of designers who can do HTML/CSS a lot faster and cheaper. Having a broader scope helps to see where you fit in, ie starting as “full stack” but then finding more satisfaction in backend

> it’s way easier to start to learn js in the browser, minimal markup to load your js and done. 
> Check everything in the console, don’t worry about packages, installers etc, just to learn the basics.

> learning how to mess with the markup, where to put the JS in the markup (<script />), etc ... feels harder to me than node

> the good thing if you code in the browser is to see what’s going on … node/pure js is boring to a lot of people because it’s just black on white

It seems that the opinions are very different, but this leads to interesting discussions.

* Thoughts *
So what are your thoughts? Regarding attracting more people into Tech, training the juniors, empower inclusivity. Please share and comment. If you would like any correction of the topic description, just notify me.

**Comment by PatrickHeneise** [comment]
The IHK/AHK (Cámara de Comercio Alemana para España)[http://www.ahk.es/es/formacion-profesional/fp-dual-en-espana/] already started dual apprenticeships (theoretical training in school + practical training on the job) for Mechanical Engineers in Barcelona with Seat. If you're interested, please send them a mail stating your interest. They're currently looking to start a program with 10 students (means 7-10 companies backing them).

**Comment by davidmonreal** [comment]
I totally agree with your approach. Web development is a vast field. Prepare (selected) students to enter the market, be productive and keep learning with senior developers around is, from our point of view, the right aproach. The theoretical + practical blend should be adopted by more companies if we want to increase the amount and level of talent in this sector.

Bravo, guys!

**Comment by SegersIan** [comment]
Some more additional information about training/school initiatives I found:

- There is a FreeCodeCamp Facebook group in Barcelona (`https://www.facebook.com/groups/free.code.camp.barcelona`) with about 500+ members.
- http://www.barcelonacodeschool.com/ (claims to do FCC meetups in addition being also a coding Bootcamp)
- https://codebar.io/ (entirely non-profit)
